src/serializer/ResidualHeapSerializer.js

```diff
@@ -699,6 +714,24 @@ export class ResidualHeapSerializer {
       }
     }
 
+    // make sure we're not serializing a class method here
+    if (val instanceof ECMAScriptSourceFunctionValue && this.residualClassMethodInstances.has(val)) {
+      let classMethodInstance = this.residualClassMethodInstances.get(val);
+      invariant(classMethodInstance);
+      // anything other than a class constructor should never go through serializeValue()
+      // so we need to log a nice error message to the user
+      if (classMethodInstance.methodType !== "constructor") {
+        let error = new CompilerDiagnostic(
+          "a class method incorrectly went through the serializeValue() code path",
+          val.$ECMAScriptCode.loc,
+          "PP0021",
+          "FatalError"
+        );
+        this.realm.handleError(error);
+        throw new FatalError();
```

Comment:
Can this happen? If it can never happen, then this should be an `invariant`, but if it can, then we should have a test for it.